src/pmproxy/src/server.c

```diff
@@ -828,6 +877,7 @@ check_proxy(uv_check_t *arg)
     struct proxy	*proxy = (struct proxy *)handle->data;
 
     flush_secure_module(proxy);
+    refresh_proxy_mem_metrics(proxy);
```

Comment:
This looks like its going to get updated once per loop iteration - that seems like *alot* ... ?
How about a stats timer instead, perhaps updating this (and similar in future) values once
every few seconds instead?

Reply:
I was just going with minimal changes if we want to take this for pcp-5.3.1. But yes it would be frequently updated. For this sort of thing it would be better if libpcp_mmv could register a fetch callback, so the value would only be computed on demand (and always be current rather than up to a few seconds old)

Reply:
> I was just going with minimal changes if we want to take this for pcp-5.3.1.

There's no rush on this AFAICS - maybe merge with those other pmproxy fixes (post-release)?

>  But yes it would be frequently updated.

Well, that's not good - definitely not appropriate for tomorrows release then.

> For this sort of thing it would be better if libpcp_mmv could register a fetch callback [...]

That's not feasible - libpcp_mmv has no knowledge of when fetch happens (by design).

A timer is the only viable option here and its just fine as an approach - you could even piggyback it onto the libpcp_mmv garbage collection timer (or perhaps add an interface to libpcp_web for registering timer-driven callbacks - an interface used by separate GC and metrics callbacks).

This code could live more neatly in libpcp_web then, instead of in the midst of not-really-related code in pmproxy.  We could trivially extend this timer-driven mechanism to also call getrusage and add accurate CPU utilization metrics too.  Hmm, in fact ... getrusage(2) seems to have rss stats, and it's platform-independent, so perhaps the one call could suffice (no vsize though - perhaps just that part becomes Linux-specific).